README
Library puTools - Version 3.1.0
General tools (string, time, etc)

Description
===========


Requirements
============






debian/changelog
metlibs-common-putools (8.1.4-1) unstable; urgency=low

  * fix build with regex
  * update debhelper compat to 11
  * use gtest if found by cmake
  * require cmake 3.10

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 14 Jul 2022 08:00:08 +0200

metlibs-common-putools (8.1.3-1) unstable; urgency=low

  * fix build for Ubuntu 20.04

 -- Alexander Bürger <alexander.buerger@met.no>  Tue, 6 Oct 2020 08:00:08 +0200

metlibs-common-putools (8.1.2-1) unstable; urgency=low

  * faster miutil::trim_remove_empty
  * faster miDate and miClock from std::string

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 1 Aug 2019 08:00:08 +0200

metlibs-common-putools (8.1.1-1) unstable; urgency=low

  * miTime: catch exceptions in miTime::format
  * miTime: bugfix, tests added

 -- Alexander Bürger <alexander.buerger@met.no>  Tue, 11 Dec 2018 08:00:08 +0100

metlibs-common-putools (8.1.0-1) unstable; urgency=low

  * miDate: format supports "%V" for week number

 -- Alexander Bürger <alexander.buerger@met.no>  Mon, 12 Nov 2018 08:00:08 +0100

metlibs-common-putools (8.0.0-1) unstable; urgency=low

  * miDate::Translations class

 -- Alexander Bürger <alexander.buerger@met.no>  Wed, 01 Nov 2017 08:00:08 +0100

metlibs-common-putools (7.1.0-2) unstable; urgency=low

  * version number in shlibs

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 12 Oct 2017 08:00:08 +0200

metlibs-common-putools (7.1.0-1) unstable; urgency=low

  * added from_latin1_to_utf8
  * code de-duplication in miDate/miTime formatting

 -- Alexander Bürger <alexander.buerger@met.no>  Mon, 21 Aug 2017 08:00:08 +0200

metlibs-common-putools (7.0.0-1) unstable; urgency=low

  * utf-8 time formatting
  * added new overloads of functions, need to increase so version

 -- Alexander Bürger <alexander.buerger@met.no>  Wed, 26 Apr 2017 08:00:08 +0200

metlibs-common-putools (6.0.0-1) unstable; urgency=low

  * CMake build
  * shared library

 -- Alexander Bürger <alexander.buerger@met.no>  Fri, 03 Mar 2017 08:00:08 +0100

metlibs-common-putools (5.1.13-1) unstable; urgency=low

  * we do not use curl, but we use boost date time

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 02 Feb 2017 08:00:08 +0100

metlibs-common-putools (5.1.11-1) unstable; urgency=low

  * fix version mismatch between puToolsVersion.h and configure.ac

 -- Alexander Bürger <alexander.buerger@met.no>  Fri, 28 Oct 2016 08:00:08 +0200

metlibs-common-putools (5.1.10-1) unstable; urgency=low

  * fix boost dependency and pkg-config file
  * updated packaging

 -- Alexander Bürger <alexander.buerger@met.no>  Fri, 28 Oct 2016 08:00:08 +0200

metlibs-common-putools (5.1.9-1) unstable; urgency=low

  * use classic locale in to_lower and to_upper
  * limit number of messages from miTime

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 25 Aug 2016 08:00:08 +0200

metlibs-common-putools (5.1.8-2) unstable; urgency=low

  * update build-depends

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 26 May 2016 09:00:09 +0100

metlibs-common-putools (5.1.8-1) unstable; urgency=low

  * TimeFilter - advanced option removed

 -- Lisbeth Bergholt <lisbeth.bergholt@met.no>  Fri, 13 May 2016 14:52:28 +0200

metlibs-common-putools (5.1.7-1) unstable; urgency=low

  * new version number, puToolsVersion.h was not updated last time

 -- Alexander Bürger <alexander.buerger@met.no>  Mon, 09 May 2016 09:00:09 +0100

metlibs-common-putools (5.1.6-1) unstable; urgency=low

  * miClock - default values 0

 -- Lisbeth Bergholt <lisbeth.bergholt@met.no>  Mon, 09 May 2016 08:54:30 +0200

metlibs-common-putools (5.1.5-1) unstable; urgency=low

  * new release version

 -- Alexander Bürger <alexander.buerger@met.no>  Mon, 10 Apr 2016 13:00:00 +0200

metlibs-common-putools (5.1.5~rc1-1) unstable; urgency=low

  * added mutil::path_ctime

 -- Alexander Bürger <alexander.buerger@met.no>  Fri, 08 Apr 2016 15:00:00 +0200

metlibs-common-putools (5.1.4-1) unstable; urgency=low

  * removed SetupParser; added TimeFilter
  * code cleanup; removed many miString

 -- Alexander Bürger <alexander.buerger@met.no>  Thu, 10 Mar 2016 14:00:00 +0100

metlibs-common-putools (5.0.4-1) unstable; urgency=low

  * setupparser changes

 -- Alexander Bürger <alexander.buerger@met.no>  Mon, 22 Feb 2016 14:30:30 +0100

metlibs-common-putools (5.0.2-2) unstable; urgency=low

  * With m4, configure.ac and rules updates

 -- Lisbeth Bergholt <lisbeth.bergholt@met.no>  Fri, 02 May 2014 10:48:37 +0200

metlibs-common-putools (5.0.2-1) unstable; urgency=low

  * Bugfix encoding (UTF-8 -> latin1)
  * miClock accepts xx:xx:xx, xx:xx, xx, xxxx and xxxxxx

 -- Lisbeth Bergholt <lisbeth.bergholt@met.no>  Fri, 02 May 2014 09:52:48 +0200

metlibs-common-putools (5.0.1-1) unstable; urgency=low

  * workaround for bdiana using static libs to build shared lib

 -- Lisbeth Bergholt <lisbeth.bergholt@met.no>  Wed, 05 Feb 2014 15:38:03 +0100

metlibs-common-putools (5.0.0-2) unstable; urgency=low

  * debian files and google tests

 -- Lisbeth Bergholt <lisbeth.bergholt@met.no>  Tue, 04 Feb 2014 15:38:06 +0100

metlibs-common-putools (5.0.0-1) unstable; urgency=low

  * Split from all-in-one metlibs

 -- Alexander Carl Friedrich Bürger <alexander.buerger@met.no>  Mon, 03 Feb 2014 09:18:04 +0100


ttycols.cc
/*
  libpuTools - Basic types/algorithms/containers

  $Id$

  Copyright (C) 2006 met.no

  Contact information:
  Norwegian Meteorological Institute
  Box 43 Blindern
  0313 OSLO
  NORWAY
  email: diana@met.no

  This library is free software; you can redistribute it and/or
  modify it under the terms of the GNU Lesser General Public
  License as published by the Free Software Foundation; either
  version 2.1 of the License, or (at your option) any later version.

  This library is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
  Lesser General Public License for more details.

  You should have received a copy of the GNU Lesser General Public
  License along with this library; if not, write to the Free Software
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
*/

#ifdef HAVE_CONFIG_H
#include "config.h"
#endif

#include "ttycols.h"

/* Created by met.no/PU
   at Fri Nov  1 10:26:17 2002 */

using namespace std;
using namespace miutil;

static const miString ttcctrl = "[";


miString ttc::foreground(ttc::color_ c)
{
  switch (c) {
  case Black:
    return "30";
  case Red :
    return "31";
  case  Green:
    return "32";
  case  Yellow:
    return "33";
  case  Blue:
    return "34";
  case  Magenta:
    return "35";
  case  Cyan:
    return "36";
  case  White:
    return "37";
  }
  return "30";
}

miString ttc::background(ttc::color_ c)
{
  switch (c) {
  case Black:
    return "40";
  case Red :
    return "41";
  case  Green:
    return "42";
  case  Yellow:
    return "43";
  case  Blue:
    return "44";
  case  Magenta:
    return "45";
  case  Cyan:
    return "46";
  case  White:
    return "47";
  }
  return "40";
}


miString ttc::attribute( ttc::attri_ att )
{
  switch (att) {
  case  Plain:
    return "00;";
  case  Bold:
    return "01;";
  case  Underscore:
    return "04;";
  case  Blink:
    return "05;";
  case  Reverse:
    return "07;";
  case  Concealed:
    return "08;";
  }
  return "00;";
}

miString ttc::color(ttc::color_ fg, ttc::attri_ att )
{
  return ttcctrl + attribute(att) + foreground(fg) + "m";
}

miString ttc::color(ttc::color_ fg, ttc::color_ bg, ttc::attri_ att)
{
  return ttcctrl+attribute(att)+foreground(fg)+";"+background(bg)+"m";
}











